### ISSUE TITLE
[java] ClassWithOnlyPrivateConstructorsShouldBeFinal - false negative when a compilation unit contains two class declarations

### ISSUE THREAD

**Issue by MetaBF** [pmd/pmd#3851]
**Affects PMD Version: 6.43.0** 

**Rule: ClassWithOnlyPrivateConstructorsShouldBeFinal** 
https://pmd.github.io/pmd-6.42.0/pmd_rules_java_design.html#classwithonlyprivateconstructorsshouldbefinal

**Description:**
In this case, `Example` is a class having only one private constructor.  According to the rule, the class should be declared final. However, PMD misses this case.

**Code Sample demonstrating the issue:**

```
public class Example { // false-negative here
    private Example() {
    }
}

class Another {
}

``` 

**Expected outcome:**

PMD should report a violation at line 1, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*

**Comment by adangel** [comment]
Thanks for the report, this is indeed a false-negative.
